File: Year-2023/LeetCode-March-Challenge/KokoEatingBananas.java
class Solution {
    public int minEatingSpeed(int[] piles, int h) {
        int low = 1, high = piles[0];
        for (int x : piles)
            high = Math.max(high, x);
        while (low <= high) {
            int mid = low + (high - low) / 2;
            long count = 0;
            for (int i = 0; i < piles.length; ++i) {
                if (piles[i] < mid)
                    count++;
                else
                    count += ((piles[i] / mid) + ((piles[i] % mid) != 0 ? 1 : 0));
            }
            if (count > h)
                low = mid + 1;
            else
                high = mid - 1;
        }
        return low;
    }
}

public class KokoEatingBananas {
    public static void main(String[] args) {
        int[] piles = { 3, 6, 7, 11 };
        int h = 8;
        Solution s = new Solution();
        System.out.println(s.minEatingSpeed(piles, h));
    }
}

File: Year-2023/LeetCode-March-Challenge/KokoEatingBananas.js
var minEatingSpeed = function(piles, h) {
    let low = 1, high = piles[0];
    for(let x of piles) high = Math.max(high, x);

    while(low <= high){
        let mid = low + Math.trunc((high - low)/2);
        let count = 0;
        for(let i = 0; i < piles.length; ++i){
            if(piles[i] < mid) count++;
            else count += (Math.trunc(piles[i]/ mid) + (Math.trunc(piles[i] % mid) !== 0));
        }
        if(count > h) low = mid + 1;
        else high = mid - 1;
    }
    return low;
};

let piles = [3,6,7,11], h = 8;
console.log(minEatingSpeed(piles, h));